library/data/registry_dicts/src/main/java/org/quiltmc/qsl/registry/dict/impl/reloader/DictMap.java

```diff
@@ -0,0 +1,231 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.dict.impl.reloader;
+
+import static org.quiltmc.qsl.registry.dict.impl.reloader.RegistryDictReloader.LOGGER;
+
+import java.io.InputStreamReader;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import com.google.gson.JsonSyntaxException;
+import com.mojang.serialization.DataResult;
+import com.mojang.serialization.JsonOps;
+import org.apache.logging.log4j.Level;
+
+import net.minecraft.resource.Resource;
+import net.minecraft.util.Identifier;
+import net.minecraft.util.InvalidIdentifierException;
+import net.minecraft.util.JsonHelper;
+import net.minecraft.util.registry.Registry;
+
+import org.quiltmc.qsl.registry.dict.api.RegistryDict;
+
+final class DictMap {
+	private final Registry<?> registry;
+	private final RegistryDict<?, ?> dictionary;
+	private final TagGetter tagGetter;
+	private final Map<DictTarget, Object> map;
```

Comment:
Is there a good reason this class has zero type safety?

Reply:
None that I recall!

Reply:
Then I'd definitely like to see this get generified prior to merge.